### PR TITLE
JAMES-3454 Use a callback mechanism to re-create RabbitMQ auto-delete queues upon reconnections

### PR DESCRIPTION
**Pull request by chibenwa**
Requires manual testing on DEV before adoption

### CODE REVIEW

**Comment by chibenwa** [comment]
Tested manually on DEV, no James reboot required on DEV

File: server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/CancelRequestQueueName.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package org.apache.james.task.eventsourcing.distributed;
```
Comment (chibenwa) [inline server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/CancelRequestQueueName.java:1]: License!!!

File: server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueueReconnectionHandler.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package org.apache.james.task.eventsourcing.distributed;
```
Comment (chibenwa) [inline server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueueReconnectionHandler.java:1]: License!!!

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**